embedchain/embedchain.py

```diff
@@ -615,13 +623,25 @@ def chat(
             logger.debug("Cache disabled. Running chat without cache.")
             if self.llm.config.token_usage:
                 answer, token_info = self.llm.query(
-                    input_query=input_query, contexts=contexts_data_for_llm_query, config=config, dry_run=dry_run
+                    input_query=input_query,
+                    contexts=contexts_data_for_llm_query,
+                    config=config,
+                    dry_run=dry_run,
+                    memories=memories,
                 )
             else:
                 answer = self.llm.query(
-                    input_query=input_query, contexts=contexts_data_for_llm_query, config=config, dry_run=dry_run
+                    input_query=input_query,
+                    contexts=contexts_data_for_llm_query,
+                    config=config,
+                    dry_run=dry_run,
+                    memories=memories,
                 )
 
+        # Add to Mem0 memory if enabled
+        if self.mem0_client:
+            self.mem0_client.add(data=input_query, agent_id=self.config.id, session_id=session_id)
```

Comment:
We may have to change the way on how to create memories based on the user query. It is possible that directly passing the user query is not the right way and we might have to prepend it with some text like "Remember user preferences from following user query: {input_query}" but we can revisit this later after the first version release.

Reply:
Alright, I'll add a TODO there.